### ISSUE TITLE
Error after dropping invalid element over editor area

### ISSUE THREAD

**Issue by jeanp413** [microsoft/vscode#124946]
- VS Code Version: stable and from sources

Steps to Reproduce:
![editor_invalid_drop](https://user-images.githubusercontent.com/25115070/120124211-e52b7980-c178-11eb-99a6-7dbe6cb0815e.gif)

**Comment by bpasero** [comment]
It looks like the raw dragged data is a UUID (e.g. `"1fa3ce57-ce74-4ac8-a729-bb4d850eae15"`) and calling `URI.parse` on such a string returns a file URI, e.g. this:

```ts
URI.parse("foo")
```

results in this:

```ts
Uri {scheme: "file", authority: "", path: "/foo", query: "", fragment: "", …}
```

@isidorn fyi if that is intentional and @jrieken whether this is expected from the URI and whether I can figure this out upfront.

My workaround would be to ignore any strings that do not contain a `:` which hints at a string that does not seem to be a URI.

**Comment by isidorn** [comment]
@bpasero yes, we set the uri on purpose [here](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/watchExpressionsView.ts#L321). By the specs of the tree I need to have some uri for my element so it is draggable and that I can identify which element it is once it is dropped. I can of course change that URI to be something else if that will help.

**Comment by bpasero** [comment]
Ok understood, I now do this:

https://github.com/microsoft/vscode/blob/88875325a703515051c5997fdd2787c391cf785b/src/vs/workbench/browser/dnd.ts#L79-L79

But wonder if there is something else I could do...

**Comment by jeanp413** [comment]
@bpasero  maybe we could:
1. Change the dataTransfer type in [`listview.ts` from `DataTransfers.RESOURCES`](https://github.com/microsoft/vscode/blob/66579f083e338575a97b2be346f9cf6c17ce9a1a/src/vs/base/browser/ui/list/listView.ts#L964) to some other more generic type so `DataTransfers.RESOURCES` is [used only for file resources](https://github.com/microsoft/vscode/blob/66579f083e338575a97b2be346f9cf6c17ce9a1a/src/vs/workbench/browser/dnd.ts#L286)
2. Validate the [dataTransfer types in EditorDropTarget::onDragEnter](https://github.com/microsoft/vscode/blob/66579f083e338575a97b2be346f9cf6c17ce9a1a/src/vs/workbench/browser/parts/editor/editorDropTarget.ts#L597) with `containsDragType(event, DataTransfers.FILES, CodeDataTransfers.FILES, DataTransfers.RESOURCES)` so the drop overlay doesn't show up for invalid types

**Comment by isidorn** [comment]
@jeanp413 That's a good idea overall but would need adoption from all the `listview` clients, and there's quite a lot of them. fyi @joaomoreno for thoughts

**Comment by joaomoreno** [comment]
Fine by me.

**Comment by jeanp413** [comment]
Created PR #125943 fixing this.

I changed the dataTransfer type from `DataTransfers.RESOURCES` to `DataTransfers.TEXT` in `listView.ts`, let me know if that's ok.

> That's a good idea overall but would need adoption from all the listview clients

Looking around dragAndDrop data is usually retrieved from `IDragAndDropData` instances and from `dataTransfer.files`. Am I missing something?
